sa/storage-authority.go

```diff
@@ -271,3 +343,34 @@ func (ssa *SQLStorageAuthority) FinalizeAuthorization(authz core.Authorization)
 	err = tx.Commit()
 	return
 }
+
+func (ssa *SQLStorageAuthority) AddCertificate(cert []byte) (id string, err error) {
+	tx, err := ssa.db.Begin()
+	if err != nil {
+		return
+	}
+
+	// Manually set the index, to avoid AUTOINCREMENT issues
+	var sequence int64
+	var scanTarget sql.NullInt64
+	err = tx.QueryRow("SELECT max(sequence) FROM certificates;").Scan(&scanTarget)
+	switch {
+	case !scanTarget.Valid:
+		sequence = 0
+	case err != nil:
+		tx.Rollback()
+		return
+	default:
+		sequence += scanTarget.Int64 + 1
+	}
+
+	id = core.Fingerprint256(cert)
+	_, err = tx.Exec("INSERT INTO certificates (sequence, digest, value) VALUES (?,?,?);", sequence, id, cert)
```

Comment:
Unfortunately, that doesn't work for the empty table.  That's probably an initialization condition that could be overcome.  I'm going to leave it for now.